src/utilities/decorators/withResponsiveMode.tsx

```diff
@@ -26,12 +27,15 @@ const RESPONSIVE_MAX_CONSTRAINT = [
 export function withResponsiveMode<P, S>(ComposedComponent: any): any {
 
   return class WithResponsiveMode extends React.Component<P, IWithResponsiveModeState> {
```

Comment:
if we extend BaseDecoratorComponent, we could reduce a lot of duplication by moving it there.